policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/config/RedisTtlConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.policy.config;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.core.RedisKeyValueAdapter.EnableKeyspaceEvents;
+import org.springframework.data.redis.repository.configuration.EnableRedisRepositories;
+
+import uk.gov.gchq.palisade.service.policy.PolicyApplication;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Additional Redis configuration to set time-to-live on a per-keyspace basis.
+ * This must still be enabled in the domain entity with the {@link org.springframework.data.redis.core.TimeToLive} annotation.
+ */
+@Configuration
+@ConditionalOnProperty(
+        prefix = "spring.data.redis.repositories",
+        name = "enabled",
+        havingValue = "true"
+)
+@EnableRedisRepositories(enableKeyspaceEvents = EnableKeyspaceEvents.ON_STARTUP, basePackageClasses = {PolicyApplication.class})
+@EnableConfigurationProperties(RedisTtlProperties.class)
+public class RedisTtlConfiguration {
```

Comment:
This is the persistence way of doing time-to-live, we're only using caching here so this shouldn't be needed

Reply:
removed